Code/GraphMol/ChemReactions/ReactionUtils.cpp

```diff
@@ -179,4 +179,80 @@ bool isReactionTemplateMoleculeAgent(const ROMol &mol, double agentThreshold)
   return true;
 }
 
+namespace {
+
+std::map<int,Atom::ChiralType> getMappingNumAtomIdxMapReactants(ChemicalReaction *rxn){
+  std::map<int,Atom::ChiralType> res;
+  for(MOL_SPTR_VECT::const_iterator reactIt=rxn->beginReactantTemplates();
+      reactIt!=rxn->endReactantTemplates();++reactIt){
+    for(ROMol::AtomIterator reactAtomIt=(*reactIt)->beginAtoms();
+        reactAtomIt!=(*reactIt)->endAtoms();++reactAtomIt){
+      if(!(*reactAtomIt)->hasProp("molAtomMapNumber")){
+        continue;
+      }
+      int reactMapNum;
+      (*reactAtomIt)->getProp("molAtomMapNumber",reactMapNum);
+      res.insert(std::make_pair(reactMapNum,(*reactAtomIt)->getChiralTag()));
+    }
+  }
+  return res;
+}
+
+}
+
+void updateProductsStereochem(ChemicalReaction *rxn)
+{
+  std::map<int,Atom::ChiralType> reactantMapping = getMappingNumAtomIdxMapReactants(rxn);
```

Comment:
Minor efficiency thing: wouldn't it make more sense to have getMappingNumAtomIdxMapReactants() take a reference to the std::map as an argument and thus save yourself the copy that's done by returning it?